core/src/main/java/hudson/model/Run.java

```diff
@@ -1911,7 +1913,7 @@ protected final void execute(@NonNull RunExecution job) {
                 }
 
                 // even if the main build fails fatally, try to run post build processing
-                job.post(listener);
+                job.post(Objects.requireNonNull(listener));
```

Comment:
This is always non-null, but the static analyzer can't figure it out. Adding this call helps the static analyzer and makes the code fail fast if a future refactoring breaks the existing invariant.